docker/build/cudaq.dev.Dockerfile

```diff
@@ -35,6 +35,13 @@ ARG destination="$CUDAQ_REPO_ROOT"
 ADD "$workspace" "$destination"
 WORKDIR "$destination"
 
+# mpich or openmpi
+ARG mpi=
+RUN if [ -n "$mpi" ]; \
+    then \
+        apt update && apt install -y lib$mpi-dev ; \
```

Comment:
Maybe double check if MPI_PATH is already defined (the case if we use the extended dev dependencies), and fail the build if both MPI_PATH and the mpi arg are defined. It seems that we don't really intend this to be used when MPI is already installed in the image, so better to actually fail that case than to have it behave in ways we did not intend it to.